e2etests/test/module_test_helper.js

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var uuid = require('uuid');
+var debug = require('debug')('e2etests:module-test-helper');
+var async = require('async');
+
+var Registry = require('azure-iothub').Registry;
+var ConnectionString = require('azure-iot-common').ConnectionString;
+var DeviceClient = require('azure-iot-device').Client;
+var ServiceClient = require('azure-iothub').Client;
+var DeviceIdentityHelper = require('./device_identity_helper.js');
+
+var hubConnectionString = process.env.IOTHUB_CONNECTION_STRING;
+var registry = Registry.fromConnectionString(hubConnectionString);
+var serviceClient = ServiceClient.fromConnectionString(hubConnectionString);
+
+module.exports.createModule = function(testModule, Transport, done)  {
+  async.series([
+    function createDeviceIfNecessary(done) {
+      if (testModule.testDevice) {
+        done();
+      } else {
+        testModule.registry = registry;
+        testModule.serviceClient = serviceClient;
+
+        debug('Creating SAS device to use with test');
+        DeviceIdentityHelper.createDeviceWithSas(function (err, testDevice) {
+          debug('createDeviceWithSas returned ' + (err ? err : 'success'));
+          if (err) {
+            done(err);
+          } else {
+            testModule.testDevice = testDevice;
+            testModule.deviceId = testDevice.deviceId;
+            done();
+          }
+        });
+      }
+    },
+    function createModule(done) {
+      testModule.moduleId = 'node_e2e_' + uuid.v4();
+      debug('creating module with id ' + testModule.moduleId);
+      registry.addModule({deviceId: testModule.deviceId, moduleId: testModule.moduleId}, function(err) {
+        debug('addModule returned ' + (err ? err : 'success'));
+        if (err) {
+          done(err);
+        } else {
+          debug('getting module with deviceId = ' + testModule.deviceId + ' and moduleId ' + testModule.moduleId);
+          registry.getModule(testModule.deviceId, testModule.moduleId, function(err, foundModule) {
+            debug('getModule returned ' + (err ? err : 'success'));
+            if (err) {
+              done(err);
+            } else {
+              var hubName = ConnectionString.parse(hubConnectionString).HostName;
+              testModule.moduleConnectionString = 'HostName=' + hubName + ';DeviceId=' + foundModule.deviceId + ';ModuleId='+foundModule.moduleId+';SharedAccessKey=' + foundModule.authentication.symmetricKey.primaryKey;
```

Comment:
>moduleConnectionString [](start = 25, length = 22)

maybe for later, maybe for now: we should have an API in the SDK that given these parameters create the connection String for us.

Reply:
that would be neat, but, later...

---
In reply to: [182244073](https://github.com/Azure/azure-iot-sdk-node/pull/271#discussion_r182244073) [](ancestors = 182244073)